fili-core/src/test/groovy/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProviderSpec.groovy

```diff
@@ -138,86 +112,37 @@ class LuceneSearchProviderSpec extends SearchProviderSpec<LuceneSearchProvider>
         1 * keyValueStore.put('cardinality_key', '14')
     }
 
-    def "moveDirEntries moves all entries of a directory to a new directory, while keeping all old empty dirs"() {
-        expect:
-        Files.exists(sourcePath)
-        Files.exists(file1)
-        Files.exists(file2)
-        Files.exists(file3)
-        Files.exists(subDir)
-        Files.exists(file4)
-
-        !Files.exists(destinationPath.resolve("segments_1"))
-        !Files.exists(destinationPath.resolve("_1.cfs"))
-        !Files.exists(destinationPath.resolve("_1.si"))
-        !Files.exists(destinationPath.resolve("subDir"))
-        !Files.exists(destinationPath.resolve("subDir").resolve("subDirFile"))
+    def "replaceIndex hot-swaps Lucene indexes in place"() {
```

Comment:
It's hard to read this core change of this PR with interleaving red & green lines, but here is what you will see if you check out this PR:

```groovy
    def "replaceIndex hot-swaps Lucene indexes in place"() {
        given: "a set of new index files and old index files"
        String sourceDir = "target/tmp/dimensionCache/animal/new_lucene_indexes"
        String destinationDir = "target/tmp/dimensionCache/animal/lucene_indexes"

        Path sourcePath = Files.createDirectories(new File(sourceDir).getAbsoluteFile().toPath())
        Path destinationPath = Files.createDirectories(new File(destinationDir).getAbsoluteFile().toPath())

        List<Path> newIndexFiles = generateIndexFiles(sourcePath)
        List<Path> oldIndexFiles = Files.list(destinationPath).collect(Collectors.toList())

        expect: "new index files do not exist in lucene directory"
        newIndexFiles.forEach{it -> !Files.exists(it)}

        and: "old index files exist in lucene directory"
        oldIndexFiles.forEach{it -> Files.exists(it)}

        when: "we swap the new index files with the old index files"
        searchProvider.replaceIndex(sourceDir)

        then: "the path of lucene directory remains unchanged"
        Files.exists(destinationPath)

        and: "new index files are in the lucene directory"
        newIndexFiles.forEach{it -> Files.exists(it)}

        and: "old index files are gone"
        oldIndexFiles.forEach{it -> !Files.exists(it)}

        and: "directory that used to contain the new index files is also gone"
        !Files.exists(sourcePath)
    }
```